Makefile

```diff
@@ -2,7 +2,7 @@ prefix := /usr/local
 bindir = $(prefix)/bin
 srcdir := $(dir $(lastword $(MAKEFILE_LIST)))
 FUT_HOST = cpp
-CXXFLAGS = -Wall -O2
+CXXFLAGS = -Wall -Wno-sign-compare -O2
```

Comment:
The original Fusion code uses `int` so the transpiler just translated them to `int` in the generated C++ code. This is my guess. Needs @pfusik to confirm this. If that's the case, then the only thing you can do is either ignore the warnings or tell the compiler not to issue them.

Reply:
I just took a quick look at the original Fusion code. It seems my guess is right.

Reply:
The other, proper alternative would be to have the Fusion code written in a way that the transpiler can produce proper type conversions in the target language (size_t/unsigned) in this case. Yes. That's more work, but not going that extra mile can lead to all sorts of issues related to [CWE-681](https://cwe.mitre.org/data/definitions/681.html) and its derivatives. I'm well aware that this likely is going to cause some breakage all over the place, but the more code is written, the worse it will be trying to fix this later.

Speaking of `int` in generated C++ code: Arrays and other data structures with more than 2^31 entries are a thing. ;-) Also, `size_t` has been around for quite a while in C (so long in fact, that I had a hard time reliably finding which of the old standards first introduced it proper). Transpiling to C (and C++) requires proper handling of signedness (even though most cases are likely to look fine). If you want to spare the overall handing of signed/unsigned API interfacing, generate proper wrappers that do checking and conversions of values passed between Fusion and C. A type system conveys some semantic in the choice of data types and a transpiler should be aware of that even if it does not always express such details (if Fusion code always uses int for array indices that's fine, but as C/C++ uses `size_t` in those cases, that's something the transpiler needs to respect when generating code).